exporters/tests/test_deserialization.py

```diff
@@ -9,12 +9,77 @@
     FieldTypeCheckError,
     InnerFieldDeserializationErrors,
     MissingFieldError,
+    _extract_dict_types,
+    _extract_list_type,
+    _extract_optional_type,
     deserialize,
     nested,
     retain_source,
 )
 
 
+@pytest.mark.parametrize(
+    "type_",
+    [list[int], str, set[str], Iterable[str], Container[str], tuple[float, float]],
+)  # Iterable and container should not have __getitem__, tuple should pass?
+def test_extract_dict_types_from_non_dict(type_: type):
```

Comment:
I don't think we should test implementation details here. This should be tested as part of the regular testing.

There's this idea of "the testing pyramid", where:
-  integration tests are large, harder to write (and sometimes maintain), but very valuable even for one test.
-  unit tests are small, simple, and easy to maintain, but less valuable individually.

The deserialization unit tests for classes are "bigger" unit tests, and will hit this code path regardless.

Reply:
this one I was trying to hit one of the ifs of the function (And I was only able with set), but I will remove the more unusual types